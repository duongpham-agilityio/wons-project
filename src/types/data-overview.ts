export interface OverviewDataItem {
  value: string;
  label: string;
  growth: string;
}
