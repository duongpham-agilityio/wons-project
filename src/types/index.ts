export * from './IconProps';
export * from './products';
