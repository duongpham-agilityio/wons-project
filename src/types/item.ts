export interface Item {
  key: string;
  label: string;
}
