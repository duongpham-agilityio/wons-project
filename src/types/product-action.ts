export interface ProductAction {
  key: string;
  label: JSX.Element | string;
  color: 'default' | 'danger' | 'primary' | 'secondary' | 'success' | 'warning';
  className?: string;
  onOpen?: () => void;
}
