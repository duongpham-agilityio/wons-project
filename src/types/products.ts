export interface IProducts {
  key: string;
  name: string;
  status: string;
  views: string;
  sales: string;
  conversion: string;
  total: string;
  img: string[];
  id: string;
  fallbackSrc: string;
  file: string[];
  download: string;
  link: string;
  personal: string;
}

export interface ProductList {
  products: IProducts[];
}

export interface TableRowCustomProps {
  item: IProducts;
  index: number;
  currentPage: number;
  itemsPerPage: number;
}
