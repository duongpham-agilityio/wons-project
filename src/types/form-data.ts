import { ReactNode } from 'react';
import { IProducts } from './products';

export interface FormData {
  name: string;
  description: string;
  price: string;
  view: string;
  sales: string;
  conversion: string;
  download: string;
  productLink: string;
  personalNote: string;
}

export interface FormProps {
  data?: IProducts | null;
  modePage: 'add' | 'edit' | 'detail';
  label: string | ReactNode;
  onSubmit?: (formData: IProducts) => void;
  id?: string;
}
