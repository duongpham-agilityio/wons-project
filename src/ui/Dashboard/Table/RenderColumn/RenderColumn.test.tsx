import { render } from '@testing-library/react';
import { RenderColumn } from '..';
import { TableRowCustomProps } from '@/types';

describe('renderColumn', () => {
  const baseProps = {
    item: {
      key: '1',
      name: 'Product 1',
      status: 'Active',
      views: '100',
      sales: '10',
      conversion: '5%',
      total: '$100',
    },
    index: 0,
    currentPage: 1,
    itemsPerPage: 10,
  };

  test('renders column "no" correctly', () => {
    const { container } = render(
      RenderColumn('no', baseProps as TableRowCustomProps),
    );
    expect(container).toMatchSnapshot();
  });

  test('renders column "name" correctly', () => {
    const { container } = render(
      RenderColumn('name', baseProps as TableRowCustomProps),
    );
    expect(container).toMatchSnapshot();
  });

  test('renders column "img" correctly', () => {
    const { container } = render(
      RenderColumn('img', baseProps as TableRowCustomProps),
    );
    expect(container).toMatchSnapshot();
  });

  test('renders column "status" correctly', () => {
    const { container } = render(
      RenderColumn('status', baseProps as TableRowCustomProps),
    );
    expect(container).toMatchSnapshot();
  });

  test('renders column "details" correctly', () => {
    const { container } = render(
      RenderColumn('details', baseProps as TableRowCustomProps),
    );
    expect(container).toMatchSnapshot();
  });

  test('renders default case for column correctly', () => {
    const { container } = render(
      RenderColumn('unknown', baseProps as TableRowCustomProps),
    );
    expect(container).toMatchSnapshot();
  });
});
