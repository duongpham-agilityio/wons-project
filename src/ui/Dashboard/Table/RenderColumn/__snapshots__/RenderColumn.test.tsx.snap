// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderColumn renders column "details" correctly 1`] = `<div />`;

exports[`renderColumn renders column "img" correctly 1`] = `<div />`;

exports[`renderColumn renders column "name" correctly 1`] = `<div />`;

exports[`renderColumn renders column "no" correctly 1`] = `<div />`;

exports[`renderColumn renders column "status" correctly 1`] = `<div />`;

exports[`renderColumn renders default case for column correctly 1`] = `<div />`;
